bin/circle/search-circle-builds

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env ruby
+# A script to analyze a range of CircleCI builds for known failures.
+# For available options, run ./search-circle-builds -h
+
+require 'pp'
+require 'json'
+require 'memoist'
+require 'open-uri'
+require 'optparse'
+
+CIRCLE_API_BASE = 'https://circleci.com/api/v1.1/project/github/code-dot-org/code-dot-org'
+
+KNOWN_FAIL_TYPES = [
+    {
+        search_string: 'ConnectionTimeoutError',
+        step_string: 'run_tests',
+        container: 1
+    },
+    {
+        search_string: 'io_fillbuf',
+        step_string: 'run_tests',
+        container: 1
+    },
+]
```

Comment:
This tool seems like it would be more useful if it grepped for whatever string I specify in the `run_tests` step of every container. What's the idea behind using a hard-coded list here instead? Perhaps if the KNOWN_FAIL_TYPES is useful also, this could be factored such that either usage is possible?